### PR TITLE
feat(apps/prod/tekton/configs): add task and triggers for auto sync OWNERS for hotfix branches

### PR DESCRIPTION
**Pull request by wuhuizuo**
Automatically create pull requests when hot fix branch is created:

1. Setup the Tekton triggers and tasks to watch the repositories on branch creating events.
2. Call the Deno script to create pull request to the target branch if the matched branch is created.

### CODE REVIEW

File: apps/prod/tekton/configs/triggers/bindings/github-branch-create.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ spec:
     - name: git-url
       value: $(body.repository.url)
     - name: git-ref
-      value: $(body.ref)
+      value: $(body.ref.substringAfter('refs/heads/')) # "refs/heads/xxx" => "xxx"
```
Comment (wuhuizuo) [inline apps/prod/tekton/configs/triggers/bindings/github-branch-create.yaml:10]:
```suggestion
      value: $(body.ref.substringAfter('refs/heads/')) # "refs/heads/xxx" => "xxx"
```

File: apps/prod/tekton/configs/triggers/templates/branch-create-product-components-hotfix.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          "tekton.dev/status-target-url":
+            "https://do.pingcap.net/tekton/#/namespaces/{{ .Namespace }}/pipelineruns/{{ .Name }}"
+          "tekton.dev/git-repo": $(tt.params.git-url)
+          "tekton.dev/git-revision": $(tt.params.git-revision)
+      spec:
+        params:
+          - name: git-url
+            value: $(tt.params.git-url)
+          - name: branch
+            value: git-ref
```
Comment (wuhuizuo) [inline apps/prod/tekton/configs/triggers/templates/branch-create-product-components-hotfix.yaml:28]:
```suggestion
            value: $(tt.params.git-ref)
```

File: apps/prod/tekton/configs/triggers/triggers/create-branch-product-components-hotfix.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+              'tidb',
+              'tiflash',
+              'tiflow',
+              'tidb-binlog',
+              'tidb-dashboard',
+              'tidb-tools'
+            ]
+    # filter git refs that match regexp: ^refs/heads/release-\d+\.\d+-\d+-v\d+\.\d+\.\d+$
+    # example values of `body.ref`:
+    #   refs/heads/release-6.5-20231020-v6.5.5
```
Comment (purelind) [inline apps/prod/tekton/configs/triggers/triggers/create-branch-product-components-hotfix.yaml:25]: Need update regexp to match `refs/heads/release-6.5-20231020-v6.5.5-2` or `refs/heads/release-6.5-20231020-v6.5.5-3`

Comment (wuhuizuo) [inline apps/prod/tekton/configs/triggers/triggers/create-branch-product-components-hotfix.yaml:25]:
```suggestion
    #   refs/heads/release-6.5-20231020-v6.5.5
     #   refs/heads/release-6.5-20231020-v6.5.5-1

```

File: apps/prod/tekton/configs/triggers/triggers/create-branch-product-components-hotfix.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    #   refs/heads/release-6.5-20231020-v6.5.5-1
+    - name: filter on branch names
+      ref:
+        name: cel
+      params:
+        - name: filter
+          value: >-
+            body.ref.matches('^refs/heads/release-\\d+\\.\\d+-\\d+-v\\d+\\.\\d+\\.\\d+$')
```
Comment (wuhuizuo) [inline apps/prod/tekton/configs/triggers/triggers/create-branch-product-components-hotfix.yaml:33]:
```suggestion
            body.ref.matches('^refs/heads/release-\\d+\\.\\d+-\\d+-v\\d+\\.\\d+\\.\\d+(-.+)?$')
            
```

Comment (wuhuizuo) [reply apps/prod/tekton/configs/triggers/triggers/create-branch-product-components-hotfix.yaml:33]: Fixed and tested

**Review by purelind — APPROVED** [review]
/lgtm

**Pull request merged**